src/routes/index.es

```diff
@@ -1,27 +1,27 @@
 import memoize from 'lodash/memoize';
 import { stringify, parse } from 'qs';
 
-const REGEX_INT = '\\d+';
-const REGEX_SLUG = '[\\w-]+';
+
+const PARAM_TOKEN = /(:\w+)\[(\w+)\]/g;
 
 export const validations = {
-  id: REGEX_INT,
-  hoodId: REGEX_INT,
+  id: '\\d+',
   zipCode: '\\d{5}',
   token: '[\\w-]{20}',
   accessCode: '[\\w]{5}-[\\w]{5}',
-  slug: REGEX_SLUG,
-  hood: REGEX_SLUG,
+  slug: '[\\w-]+',
   stringId: '\\w{6,}',
 };
 
-export const getParamReplacement = memoize((fragment, param) => {
-  const regex = validations[param];
-  if (!regex) return fragment;
-  return `${fragment}(${regex})`;
+export const getParamReplacer = (typesValidation) => memoize((fragment, param, type) => {
+  const regex = typesValidation[type];
+  return regex ? `${param}(${regex})` : fragment;
```

Comment:
If no regex is found but requested this will return a broken path: ‘/test/:param[nonexistent]’